resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -16,256 +16,130 @@
-      .provideInitContainerBundle(maybeSubmittedResourceIdentifiers.map(_.ids()),
-        resolvedSparkJars ++ resolvedSparkFiles)
-
-    val podWithInitContainer = initContainerBundler.map(
-      _.sparkPodInitContainerBootstrap
-        .bootstrapInitContainerAndVolumes(driverContainer.getName, basePod))
-      .getOrElse(basePod)
-    val sparkConfWithExecutorInit = initContainerBundler.map(
-      _.executorInitContainerConfiguration
-        .configureSparkConfForExecutorInitContainer(sparkConf))
-      .getOrElse(sparkConf)
-    val credentialsMounter = kubernetesCredentialsMounterProvider
-        .getDriverPodKubernetesCredentialsMounter()
-    val credentialsSecret = credentialsMounter.createCredentialsSecret()
-    val podWithInitContainerAndMountedCreds = credentialsMounter.mountDriverKubernetesCredentials(
-      podWithInitContainer, driverContainer.getName, credentialsSecret)
-    val resolvedSparkConf = credentialsMounter.setDriverPodKubernetesCredentialLocations(
-        sparkConfWithExecutorInit)
-    if (resolvedSparkJars.nonEmpty) {
-      resolvedSparkConf.set("spark.jars", resolvedSparkJars.mkString(","))
-    }
-    if (resolvedSparkFiles.nonEmpty) {
-      resolvedSparkConf.set("spark.files", resolvedSparkFiles.mkString(","))
-    }
-    resolvedSparkConf.setIfMissing(KUBERNETES_DRIVER_POD_NAME, kubernetesDriverPodName)
-    resolvedSparkConf.set("spark.app.id", kubernetesAppId)
-    resolvedSparkConf.set(KUBERNETES_EXECUTOR_POD_NAME_PREFIX, kubernetesResourceNamePrefix)
-    // We don't need this anymore since we just set the JVM options on the environment
-    resolvedSparkConf.remove(org.apache.spark.internal.config.DRIVER_JAVA_OPTIONS)
-    val resolvedLocalClasspath = containerLocalizedFilesResolver
-      .resolveSubmittedAndRemoteSparkJars()
-    val resolvedDriverJavaOpts = resolvedSparkConf.getAll.map {
-      case (confKey, confValue) => s"-D$confKey=$confValue"
-    }.mkString(" ") + driverJavaOptions.map(" " + _).getOrElse("")
-    val resolvedDriverPodBuilder = podWithInitContainerAndMountedCreds.editSpec()
-      .editMatchingContainer(new ContainerNameEqualityPredicate(driverContainer.getName))
-        .addNewEnv()
-          .withName(ENV_MOUNTED_CLASSPATH)
-          .withValue(resolvedLocalClasspath.mkString(File.pathSeparator))
-          .endEnv()
-        .addNewEnv()
-          .withName(ENV_DRIVER_JAVA_OPTS)
-          .withValue(resolvedDriverJavaOpts)
-          .endEnv()
-        .endContainer()
-      .endSpec()
-    val driverPodFileMounter = initContainerComponentsProvider.provideDriverPodFileMounter()
-    val resolvedDriverPod = pythonResource.map {
-      p => p.driverPodWithPySparkEnvs(
-        driverPodFileMounter,
-        resolvedPrimaryPySparkResource,
-        resolvedPySparkFiles.mkString(","),
-        driverContainer.getName,
-        resolvedDriverPodBuilder
-      )}.getOrElse(resolvedDriverPodBuilder.build())
+      .build()
     Utils.tryWithResource(
```

Comment:
@ash211 @mccheah I don't see overly-complex logic in the client class that would make it sensible to move the DriverSpec logic into a separate step. I understand your point Andrew, but it might not be wise to overdo the step based architecture at the cost of, as Matt said, just to move complexity from one class to another.